frontend/src/api/actions/init.js

```diff
@@ -0,0 +1,154 @@
+import { setActiveSession, fetchSessions, sessionsSelector } from "./sessions";
+import { fetchActiveUser, setActiveUserRole } from "./users";
+import { fetchApplicants } from "./applicants";
+import { fetchApplications } from "./applications";
+import { fetchAssignments } from "./assignments";
+import { fetchContractTemplates } from "./contract_templates";
+import { fetchInstructors } from "./instructors";
+import { fetchPositions } from "./positions";
+import { setGlobals, globalsSelector } from "./globals";
+
+/**
+ * A helper function to replace all API actions to
+ * use mock API actions.
+ *
+ * @param {boolean} enableMockAPI
+ */
+function toggleMockApi(enableMockAPI) {
+    // in production, ToggleMockApi is a no-op. In development, it actually
+    // does something.
+    /* eslint-disable */
+    if (enableMockAPI != null && process.env.REACT_APP_DEV_FEATURES) {
+        const mock = require("../mockAPI");
+        const mockAPI = mock.mockAPI;
+
+        if (enableMockAPI === true) {
+            mockAPI.replaceGlobalFetch();
+        } else {
+            mockAPI.restoreGlobalFetch();
+        }
+    }
+    /* eslint-enable */
+}
+
+/**
+ * A helper function to update all necessary global variables
+ *
+ * @param {bool} enableMockAPI
+ * @param {object} newActiveSession
+ * @param {function} getState
+ * @returns
+ */
+function updateGlobals(enableMockAPI, newActiveSession, getState) {
+    const state = getState();
+    let globals = globalsSelector(state);
+
+    if (enableMockAPI != null) {
+        globals = {
+            ...globals,
+            mockAPI: enableMockAPI
+        };
+    }
+
+    if (newActiveSession) {
+        globals = {
+            ...globals,
+            activeSession: newActiveSession.id
+        };
+    }
+
+    return setGlobals(globals);
+}
+
+/**
+ * A helper function to determine if the `currentStage`
+ * should be run
+ *
+ * @param {string} fromStage
+ * @param {boolean} currentStage
+ * @returns
+ */
+function shouldRunStage(fromStage, currentStage) {
```

Comment:
Why don't you move this to an inline function in `initFromStage` so that `fromStage` is defined via a closure?